presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -445,6 +445,10 @@ public static Slice multiply(Slice left, Slice right)
 
     public static void multiply(Slice left, Slice right, Slice result)
     {
+        boolean fourIntsResultExpected = result.length() == NUMBER_OF_LONGS * Long.BYTES;
```

Comment:
We need multiply that support 32bytes result only in division. I just didn't want to duplicate the multiplication code.